### ISSUE TITLE
Snapshotter container fails in k8s e2e

### ISSUE THREAD

**Issue by msau42** [kubernetes-sigs/gcp-compute-persistent-disk-csi-driver#133]
Because CRD is not installed

**Comment by gnufied** [comment]
Yeah. Seeing same. The controller errors out because snapshot CRDs aren't installed. Can we disable snapshot controller until we fix it?

**Comment by msau42** [comment]
Do you see it causing the test to fail?  Afaik, as long as the test is not testing snapshots, it should be fine

**Comment by gnufied** [comment]
No I haven't seen the test fail yet. Just the status of pod looks iffy when you list via `kubectl` .

**Comment by davidz627** [comment]
Log from the csi-snapshotter container:
```
I1101 01:59:46.925054       1 main.go:76] Version: v0.4.0-4-ge6d7b9ac
F1101 01:59:46.957513       1 create_crd.go:46] failed to create VolumeSnapshotResource: &v1beta1.CustomResourceDefinition{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1beta1.CustomResourceDefinitionSpec{Group:"", Version:"", Names:v1beta1.CustomResourceDefinitionNames{Plural:"", Singular:"", ShortNames:[]string(nil), Kind:"", ListKind:"", Categories:[]string(nil)}, Scope:"", Validation:(*v1beta1.CustomResourceValidation)(nil), Subresources:(*v1beta1.CustomResourceSubresources)(nil), Versions:[]v1beta1.CustomResourceDefinitionVersion(nil), AdditionalPrinterColumns:[]v1beta1.CustomResourceColumnDefinition(nil)}, Status:v1beta1.CustomResourceDefinitionStatus{Conditions:[]v1beta1.CustomResourceDefinitionCondition(nil), AcceptedNames:v1beta1.CustomResourceDefinitionNames{Plural:"", Singular:"", ShortNames:[]string(nil), Kind:"", ListKind:"", Categories:[]string(nil)}, StoredVersions:[]string(nil)}}, err: &errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:""}, Status:"Failure", Message:"customresourcedefinitions.apiextensions.k8s.io is forbidden: User \"system:serviceaccount:default:csi-controller-sa\" cannot create resource \"customresourcedefinitions\" in API group \"apiextensions.k8s.io\" at the cluster scope", Reason:"Forbidden", Details:(*v1.StatusDetails)(0xc420082360), Code:403}}
```
at least for the dev driver it looks like we don't have the right API permissions? The stable driver should not have the csi-snapshotter container.

/assign @jingxu97

**Comment by davidz627** [comment]
/assign @jingxu97 
third attemp

**Comment by liyinan926** [comment]
Ran into the same issue on a v1.13.0-alpha.3 cluster installed on GCE, with the dev version. Looks like the service account `csi-controller-sa` does not have permission to create `CustomResourceDefinition`s.

**Comment by davidz627** [comment]
Fixed by: #156 
/close
